File: 1306-minimum-absolute-difference/minimum-absolute-difference.ts
function minimumAbsDifference(arr: number[]): number[][] {
    const nums: number[] = arr.sort((a, b) => a - b);
    let minAbs: number = Infinity;

    for (let i = 0; i < nums.length - 1; i++) {
        const num1 = nums[i];
        const num2 = nums[i + 1];
        minAbs = Math.min(minAbs, Math.abs(num1 - num2));
    }

    const res: number[][] = [];

    for (let i = 0; i < nums.length - 1; i++) {
        const num1 = nums[i];
        const num2 = nums[i + 1];
        
        if (Math.abs(num1 - num2) === minAbs) res.push([num1, num2])
    }

    return res;
};

File: 1306-minimum-absolute-difference/minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        nums = sorted(arr)
        minAbs = float("inf")

        for i in range(len(nums) - 1):
            num1, num2 = nums[i], nums[i + 1]
            minAbs = min(minAbs, abs(num1 - num2))
        
        res = []

        for i in range(len(nums) - 1):
            num1, num2 = nums[i], nums[i + 1]
            if (abs(num1 - num2) == minAbs):
                res.append([num1, num2])
        
        return res